numba/tests/test_warnings.py

```diff
@@ -204,6 +205,23 @@ def test_disable_performance_warnings(self):
         out, err = popen.communicate()
         self.assertEqual(popen.returncode, not_found_ret_code)
 
+    def test_warning_vectorize_with_nogil(self):
+        @vectorize(nogil=True)
```

Comment:
Would be good to check `guvectorize` as well given the error message mentions it?